Microsoft.Identity.Web/Client/TokenAcquisition.cs

```diff
@@ -128,7 +128,7 @@ public async Task AddAccountToCacheFromAuthorizationCode(AuthorizationCodeReceiv
                 // even if it's not done yet, so that it does not concurrently call the Token endpoint.
                 context.HandleCodeRedemption();
 
-                var application = BuildConfidentialClientApplication(context.HttpContext, context.Principal);
+                var application = GetOrBuildConfidentialClientApplication(context.HttpContext, context.Principal);
```

Comment:
nit: what about just `GetConfidentialClientApplication` because that's what's happening, as there is another method for build?